clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2789,14 +2838,16 @@ void SYCLIntegrationHeader::startKernel(StringRef KernelName,
 }
 
 void SYCLIntegrationHeader::addParamDesc(kernel_param_kind_t Kind, int Info,
-                                         unsigned Offset) {
+                                         unsigned Offset,
+                                         unsigned NumOpenCLParams) {
   auto *K = getCurKernelDesc();
   assert(K && "no kernels");
   K->Params.push_back(KernelParamDesc());
```

Comment:
```suggestion
  K->Params.push_back({Kind, Info, Offset, NumOpenCLParams});
```

Then you can delete all lines after this one.